src/create-route-map.js

```diff
@@ -112,32 +112,40 @@ function addRouteRecord (
     })
   }
 
-  if (route.alias !== undefined) {
-    const aliases = Array.isArray(route.alias)
-      ? route.alias
-      : [route.alias]
-
-    aliases.forEach(alias => {
-      const aliasRoute = {
-        path: alias,
-        children: route.children
-      }
-      addRouteRecord(
-        pathList,
-        pathMap,
-        nameMap,
-        aliasRoute,
-        parent,
-        record.path || '/' // matchAs
-      )
-    })
-  }
-
   if (!pathMap[record.path]) {
     pathList.push(record.path)
     pathMap[record.path] = record
   }
 
+  if (route.alias !== undefined) {
+    if (route.alias === path) {
```

Comment:
The warn should only appear on dev mode. Also could you add a test for it?

Reply:
OK, I have already added